dft/src/butterflies.rs

```diff
@@ -3,14 +3,50 @@ use core::mem::MaybeUninit;
 use itertools::izip;
 use p3_field::{Field, PackedField, PackedValue};
 
+/// A butterfly operation used in NTT to combine two values into a new pair.
+///
+/// This trait defines how to transform two elements (or vectors of elements)
+/// according to the structure of a butterfly gate.
+///
+/// In an NTT, butterflies are the core units that recursively combine values
+/// across layers. Each butterfly computes:
+///
+///   (a + b * twiddle, a - b * twiddle)   // DIT
+/// or
+///   (a + b, (a - b) * twiddle)           // DIF
+///
```

Comment:
Minor fmt fix.
```suggestion
/// ```text
///   (a + b * twiddle, a - b * twiddle)   // DIT
/// or
///   (a + b, (a - b) * twiddle)           // DIF
/// ```
```